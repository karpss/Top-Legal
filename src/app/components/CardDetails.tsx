/* eslint-disable */
import React from 'react';
import { createSelector } from 'reselect';
import { useAppSelector } from '../hooks';
import { makeSelectCharactersList } from '../pages/Home/selectors';
import { Grid } from '@mui/material';
import Card from '@mui/material/Card';
import CardContent from '@mui/material/CardContent';
import CardMedia from '@mui/material/CardMedia';
import Typography from '@mui/material/Typography';
import { CardActionArea } from '@mui/material';
import Box from '@mui/material/Box';
import LinearProgress from '@mui/material/LinearProgress';

const stateSelector = createSelector(makeSelectCharactersList, (charactersList) => ({charactersList}) )

const CardDetails = () => {
  const {charactersList} = useAppSelector(stateSelector)
  return (
    <>
    {!charactersList ? (
       <Box sx={{ width: '100%' }}>
       <LinearProgress />
     </Box>
      ) : (
        <Grid container spacing={3} >
        {charactersList?.results?.map((xter) =>(
            <Grid md={2.4} lg={2} item xs={3} key={xter?.id} >
              <Card sx={{ maxWidth: 345 }}>
                <CardActionArea>
                <CardMedia
                component="img"
                height="140"
                image={xter?.image || ""}
                alt={xter?.name || ""} 
                />

                <CardContent>

          <Typography gutterBottom variant="h5" component="div">
            {xter?.name}
          </Typography>

          <Typography variant="body2" color="text.secondary">
           Status: {xter?.status}
          </Typography>

          <Typography variant="body2" color="text.secondary">
           Gender: {xter?.gender}
          </Typography>

          <Typography variant="body2" color="text.secondary">
           Location: {xter?.location?.name}
          </Typography>





                </CardContent>






                </CardActionArea>
                
                
                
                
                
                
                
                </Card>  




            </Grid>
        ))}
        
    </Grid>
     )}

    </>
  )
  
}

export default CardDetails;