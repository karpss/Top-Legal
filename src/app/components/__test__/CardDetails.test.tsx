import React from "react";
import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import { store } from "../../store";
import CardDetails from "..";

describe("Card Details Component", () => {
  it("should render CardDetails component", () => {
    const { container } = render(
      <Provider store={store}>
        <CardDetails />
      </Provider>,
    );
    expect(container).toMatchSnapshot();
  });
});
