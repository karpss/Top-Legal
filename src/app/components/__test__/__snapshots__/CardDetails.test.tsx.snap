// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component should render app component 1`] = `
<div>
  <div>
    <div
      class="MuiBox-root css-8atqhb"
    >
      <span
        class="MuiLinearProgress-root MuiLinearProgress-colorPrimary MuiLinearProgress-indeterminate css-eglki6-MuiLinearProgress-root"
        role="progressbar"
      >
        <span
          class="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Indeterminate css-5ir5xx-MuiLinearProgress-bar1"
        />
        <span
          class="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar2Indeterminate css-1r8wrcl-MuiLinearProgress-bar2"
        />
      </span>
    </div>
  </div>
</div>
`;

exports[`Card Details Component should render CardDetails component 1`] = `
<div>
  <div>
    <div
      class="MuiBox-root css-8atqhb"
    >
      <span
        class="MuiLinearProgress-root MuiLinearProgress-colorPrimary MuiLinearProgress-indeterminate css-eglki6-MuiLinearProgress-root"
        role="progressbar"
      >
        <span
          class="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Indeterminate css-5ir5xx-MuiLinearProgress-bar1"
        />
        <span
          class="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar2Indeterminate css-1r8wrcl-MuiLinearProgress-bar2"
        />
      </span>
    </div>
  </div>
</div>
`;
